frontend/src/views/SoilTests/SoilTests.tsx

```diff
@@ -29,30 +31,43 @@ import { InfoBox } from './soilTests.styles';
 import useAppState from '@/hooks/useAppState';
 import { CROPS, FIELD_LIST } from '@/constants/routes';
 import SoilTestsModal from './SoilTestsModal';
+import {
+  sharedCalcCropReq,
+  postprocessModalData,
+} from '@/calculations/FieldAndSoil/Crops/Calculations';
 
 export default function SoilTests() {
   const { state, dispatch } = useAppState();
   const navigate = useNavigate();
   const apiCache = useContext(APICacheContext);
 
-  const [fields, setFields] = useState<NMPFileField[]>(state.nmpFile.years[0].fields || []);
+  const [fields, setFields] = useState<NMPFileField[]>(
+    structuredClone(state.nmpFile.years[0].fields) || [],
```

Comment:
Same comment as above. Are we doing operations that affect the base object?

Reply:
See above.